marge/app.py

```diff
@@ -17,6 +17,16 @@
 from . import user as user_module
 
 
+
+def time_interval(s):
+    try:
+        quant, unit = re.match(r'\A([\d.]+)(h|m(?:in)?|s)?\Z', s).groups()
+        translate = {'h': 'hours', 'm': 'minutes', 'min': 'minutes', 's': 'seconds'}
+        return timedelta(**{translate[unit or 's']: float(quant)})
+    except (AttributeError, ValueError):
+        raise argparse.ArgumentTypeError('Invalid time interval (e.g. 12[s|min|h]): %s', s)
+
```

Comment:
@aschmolck Could you also allow for a space between number and unit? E.g. `2 min`. Also, since you allow for `m` and `min`, how about allowing for `sec` as well?

Reply:
The only reason I allow m(eters) for minutes is because that's unfortunately some sort of dumb pseudo standard (IIRC, grafana, prometheus, gnu sleep etc. all use `m` rather than `min`). Optional space, OTOH, I'm fine with – pushed.